server/src/ClientToDBAPI/UsersRoute/mainRoute.ts

```diff
@@ -117,6 +117,25 @@ usersRoute.post('/changeInvestigator', adminMiddleWare, (request: Request, respo
     });
 });
 
+usersRoute.post('/changeGroupInvestigator', adminMiddleWare, (request: Request, response: Response) => {
+    const changeGroupInvestigatorLogger = logger.setup({
+        workflow: 'change investigator for grouped investigatios',
+        user: response.locals.user.id,
+    });
+    const newInvestigator = request.body.user;
+    const selectedGroups = request.body.groupIds;
+    changeGroupInvestigatorLogger.info(`querying the graphql API with parameters ${JSON.stringify(request.body)}`, Severity.LOW);
+    graphqlRequest(UPDATE_INVESTIGATOR_BY_GROUP_ID, response.locals, {newInvestigator, selectedGroups})
+    .then((result: any) => {
+        changeGroupInvestigatorLogger.info(`investigator have been changed in the DB for group: ${selectedGroups}`, Severity.LOW);
+        response.send(result);
```

Comment:
add condition `if(result?.data && !result.errors)` inside the then

Reply:
fixed